scalingo/data_scalingo_region.go

```diff
@@ -0,0 +1,86 @@
+package scalingo
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/Scalingo/go-scalingo/v5"
+)
+
+func dataSourceScRegion() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceScRegionsRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"display_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"api": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dashboard": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"database_api": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ssh": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceScRegionsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _ := meta.(*scalingo.Client)
+
+	regions, err := client.RegionsList(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	name, ok := d.Get("name").(string)
+	if !ok || name == "" {
+		return diag.Errorf("name attribute is mandatory")
+	}
```

Comment:
question: just a thought I had here. As `name` is marked as `Required`, I'm pretty sure we can do:

```suggestion
	name, _ := d.Get("name").(string)
```

Don't you think? I mean, Terraform ensures that `name` is defined at this point in code, no?